### PR TITLE
eclipses: 2021-03 -> 2021-06

### PR DESCRIPTION
**Pull request by jerith666**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @mohe2015, @rycee

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 135919` at 84a751c7 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built successfully:</summary>
  <ul>
    <li>eclipses.eclipse-committers</li>
    <li>eclipses.eclipse-cpp</li>
    <li>eclipses.eclipse-java</li>
    <li>eclipses.eclipse-jee</li>
    <li>eclipses.eclipse-modeling</li>
    <li>eclipses.eclipse-platform</li>
    <li>eclipses.eclipse-rcp</li>
    <li>eclipses.eclipse-sdk</li>
  </ul>
</details>

File: pkgs/applications/editors/eclipse/plugins.nix [hunk 1]
```diff
@@ -248,12 +248,12 @@ rec {
...
     name = "cdt-${version}";
     # find current version at https://www.eclipse.org/cdt/downloads.php
-    version = "10.2.0";
+    version = "10.3.2";
 
     src = fetchzip {
       stripRoot = false;
-      url = "https://www.eclipse.org/downloads/download.php?r=1&nf=1&file=/tools/cdt/releases/10.2/${name}/${name}.zip";
-      sha256 = "1r30cbpbzw3dfcsn54p6sqip86dqhydhsppjgaz60b6z138vzx49";
+      url = "https://www.eclipse.org/downloads/download.php?r=1&nf=1&file=/tools/cdt/releases/10.3/${name}/${name}.zip";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/eclipse/plugins.nix:255]:
```suggestion
      url = "https://www.eclipse.org/downloads/download.php?r=1&nf=1&file=/tools/cdt/releases/${lib.versions.majorMinor version}/${name}/${name}.zip";
```

Comment (jerith666) [reply pkgs/applications/editors/eclipse/plugins.nix]: didn't know about that function, thanks!  fixed.

**Review by bobby285271 — APPROVED** [review]
`eclipses.eclipse-java` launches and works.

**Pull request merged**